### PR TITLE
changes for mypipeline.onnx

### PR DESCRIPTION
**Pull request by chentong319**
There are two changes for this model

1. Support the list(strings) attribute
StrArrayAttr is used. However, this attribute require arrayref to StringRef. Currently, there seems to be memory issue.

2. Fix the output type mapping.
The case in which the type is directly specified instead of using type string is added now.

### CODE REVIEW

**Comment by chentong319** [comment]
@tjingrant As for the error with StrArrayAttr, I tried to create a string object and it worked. 
` std::string *p = new std::string(item);`
These objects should be saved  and freed later, or we should provide a pool management. I feel that we also have the same memory issue with StringAttr, which uses StringRef too.

**Comment by tjingrant** [comment]
The solution to 1 needs to be updated because of the obvious memory leak issue. And my suggestion was that we should either find out how other projects manage to deal with memory storage problems induced by string array attributes or see if we can use TableGen to automatically generate something (like parser) that'll shed light on how mlir internally makes use of the string attribute storage.

As for StringAttr, I looked into the relevant code in mlir, and it appears to be fine. See how the StringAttr storage instance is created [here](https://github.com/llvm/llvm-project/blob/8f262a686e5ecd25552d6992bbccb08a8f5c3f26/mlir/lib/IR/AttributeDetail.h#L284).

And specifically the source of the StringRef is copied using a `copyInto` function defined [here](https://github.com/llvm/llvm-project/blob/cb9ae0025c4ed966a3a9b5539a9ff6b6e865516f/mlir/include/mlir/Support/StorageUniquer.h#L105).

**Comment by chentong319** [comment]
The allocate for StringAttr is for the storage of the type itself, which is for `StringRef value;`. Not the string itself.
I thought that  the string should be allocated along the onnx model. I tried to keep the `onnx::ModelProto model` global but did not fix the problem. It seems that ModelProto did not hold those strings. Maybe we can look into that part of code more. For other data from ModelProto, mlir copies the value. 
Of course, we can have a vector<std::string> for every model imported to onnx mlir and free them when the IR for the model is freed.

**Comment by chentong319** [comment]
It turned out that MLIR copied the string with its AttributeStorage. The error came from that the StringRef is made from the temporary variable for iterator. As llvm has pointed out, it requires caution to use StringRef explicitly.

File: src/Builder/FrontendDialectTransformer.cpp [hunk 1]
```diff
@@ -114,9 +114,10 @@ class FrontendGenImpl {
     case onnx::AttributeProto::INT:
       mlirAttr = builder_.getI64IntegerAttr(attr.i());
       break;
-    case onnx::AttributeProto::STRING:
-      mlirAttr = builder_.getStringAttr(attr.s());
-      break;
+    case onnx::AttributeProto::STRING: {
+      std::string *copiedString = new std::string(attr.s());
```
Comment (tjingrant) [inline src/Builder/FrontendDialectTransformer.cpp:118]: Please fix this memory leak.

**Pull request merged**